pkg/px-resource-gateway/priority_queue.go

```diff
@@ -0,0 +1,82 @@
+package server
```

Comment:
although not incorrect, it is confusing to have the package not be the name of the directory. I recommend creating a `server` directory under `px-resource-gateway` and placing these files there.